engine/engine_test.go

```diff
@@ -3399,7 +3402,12 @@ func testNativeHistograms(t *testing.T, cases []histogramTestCase, opts promql.E
 
 						sortByLabels(promResult)
 						sortByLabels(newResult)
-						testutil.Equals(t, promVector, newVector)
+						if hasNaNs(promResult) {
+							t.Log("Applying comparison with NaN equality.")
+							testutil.WithGoCmp(cmpopts.EquateNaNs()).Equals(t, promVector, newVector)
+						} else {
+							require.Equal(t, promVector, newVector)
```

Comment:
Let's maybe keep single testing lib
```suggestion
							testutil.Equals(t, promVector, newVector)
```